internal/api-key/command_delete.go

```diff
@@ -28,17 +26,38 @@ func (c *command) newDeleteCommand() *cobra.Command {
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
 	c.setKeyStoreIfNil()
-	apiKey := args[0]
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmYesNoMsg, resource.ApiKey, apiKey)
-	if ok, err := form.ConfirmDeletion(cmd, promptMsg, ""); err != nil || !ok {
+	existenceFunc := func(id string) bool {
+		_, _, err := c.V2Client.GetApiKey(id)
+		return err == nil
+	}
+
+	if err := deletion.ValidateAndConfirmDeletionYesNo(cmd, args, existenceFunc, resource.ApiKey); err != nil {
 		return err
 	}
-	httpResp, err := c.V2Client.DeleteApiKey(apiKey)
-	if err != nil {
-		return errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, httpResp)
+
+	deleteFunc := func(id string) error {
+		if r, err := c.V2Client.DeleteApiKey(id); err != nil {
+			return errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, r)
+		}
+		return nil
+	}
+
+	deletedIDs, err := deletion.Delete(args, deleteFunc, resource.ApiKey)
+
+	errs := multierror.Append(err, c.deleteKeysFromKeyStore(deletedIDs))
```

Comment:
I'm not sure we have coverage for the multiple-error case?

Reply:
I just added a multierror integration test for `environment delete`.

For the specific scenario here (appending more errors after `Delete`, it's not really something that's easily testable in integration tests because as far as I can tell, all instances where we do post-delete tasks only return errors if the user isn't logged in or if `Save()` had an error.